src/assignment-5/CollateralizedVault.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {IERC20} from "yield-utils-v2/token/IERC20.sol";
+import {AggregatorV3Interface} from "chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import {Ownable} from "openzeppelin/contracts/access/Ownable.sol";
+
+interface IERC20WithDecimals is IERC20 {
+    function decimals() external view returns (uint8);
+}
+
+/// @title Collateralized Vault
+/// @author davidbrai
+/// @notice A vault that allows to borrow a token against a collateral
+///     If a debt position is not sufficiently collateralized, e.g when the market price
+///     of the collateral goes down, then the user's collateral may be liquidated by the admin.
+///     In case of a liquidation, both the user's debt and collateral are erased.
+/// @dev The Vault uses Chainlink price feeds to determine the value of the collateral compared to the debt
+contract CollateralizedVault is Ownable {
+
+    /// @notice ERC20 token which can be borrowed from the vault
+    IERC20WithDecimals public immutable underlying;
+
+    /// @notice ERC20 token which can be used as collateral
+    IERC20WithDecimals public immutable collateral;
+
+    /// @notice Chainlink priceFeed of underlying / collateral
+    ///     e.g. if underlying is DAI and collateral is WETH, priceFeed is for DAI/WETH
+    AggregatorV3Interface public immutable priceFeed;
+
+    /// @notice deposited collateral per user in the vault, of `collateral` token
+    mapping(address => uint256) public depositedCollateral;
+
+    /// @notice Mapping of current debt per user, of `underlying` token
+    mapping(address => uint256) public debt;
```

Comment:
Not great English, but you could rename this as `borrows`, and it would follow the pattern that the mapping matches the function that increases values in it.

Reply:
👍  https://github.com/davidbrai/yield-mentorship/pull/6/commits/38956237a96e6e4a89f4f330d878632de34a1696